From 1a6d4c44b1769df0b7073691517bd1413c5344df Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Mon, 25 Nov 2019 10:38:55 +0100
Subject: [PATCH] Added SOVERSION field for shared library.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c32724b..7835c84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -345,3 +345,6 @@ if (SQLITECPP_BUILD_TESTS)
 else (SQLITECPP_BUILD_TESTS)
     message(STATUS "SQLITECPP_BUILD_TESTS OFF")
 endif (SQLITECPP_BUILD_TESTS)
+
+# API version for SQLiteCpp shared library.
+set_property(TARGET SQLiteCpp PROPERTY SOVERSION 0)
